controllers/tempo/microservices_controller.go

```diff
@@ -150,3 +179,26 @@ func isNamespaceScoped(obj client.Object) bool {
 		return true
 	}
 }
+
+func (r *MicroservicesReconciler) findMicroservicesForStorageSecret(secret client.Object) []reconcile.Request {
+	microservices := &v1alpha1.MicroservicesList{}
+	listOps := &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector(storageSecretField, secret.GetName()),
+		Namespace:     secret.GetNamespace(),
+	}
+	err := r.List(context.TODO(), microservices, listOps)
+	if err != nil {
+		return []reconcile.Request{}
+	}
+
+	requests := make([]reconcile.Request, len(microservices.Items))
+	for i, item := range microservices.Items {
+		requests[i] = reconcile.Request{
```

Comment:
Should we consider checking for changes here and trigger reconciling only if a valid change is detected?

Reply:
My envisioned workflow of this would be that on every secret change, the Reconcile function validates the secret and if the validation fails adds these errors in the status field (https://github.com/os-observability/tempo-operator/issues/153). If the validation succeeds, it updates the ConfigMap.

If relevant parts of the secret changed, the hash of the ConfigMap will be different and pods are restarted (https://github.com/os-observability/tempo-operator/issues/137). If no relevant parts of the secret changed (e.g. a user added a new key to the secret, which is not read by the operator), the checksum of the ConfigMap will stay the same and the pods are not restarted.

Reply:
(i.e. if the validation fails, Reconcile should still run and the error `Condition` should be added to the status field)